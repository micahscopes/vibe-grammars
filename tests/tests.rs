use dir_test::dir_test;
use insta::assert_snapshot;
use pest::Parser;
use vibe_grammars::{elements::{Parser as ElementsParser, Rule as ElementsRule}, util::to_sexpr};

#[dir_test(
    dir: "$CARGO_MANIFEST_DIR/tests/elements",
    glob: "chord*.txt",
)]
fn test_chords(fixture: dir_test::Fixture<&str>) {
    let input = *fixture.content();
    let result = ElementsParser::parse(ElementsRule::chords, input).unwrap();
    let result = to_sexpr(result, 0);
    let path = fixture.path();
    use std::path::Path;
    let path = Path::new(&path);
    let stem = path.file_stem().unwrap().to_str().unwrap();
    println!("Stem: {}", stem);
    assert_snapshot!(stem, result);
}