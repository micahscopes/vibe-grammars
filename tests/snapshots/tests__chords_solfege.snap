---
source: tests/tests.rs
assertion_line: 19
expression: result
---
( chords
  `Afa Bre Cdo Dla Bbso Absol G#la A#ti Fmi`
  ( chord
    `Afa`
    ( note
      `A`
      ( base
        `A`
      )
    )
    ( quality
      `fa`
      ( solfege
        `fa`
      )
    )
  )
  ( chord
    `Bre`
    ( note
      `B`
      ( base
        `B`
      )
    )
    ( quality
      `re`
      ( solfege
        `re`
      )
    )
  )
  ( chord
    `Cdo`
    ( note
      `C`
      ( base
        `C`
      )
    )
    ( quality
      `do`
      ( solfege
        `do`
      )
    )
  )
  ( chord
    `Dla`
    ( note
      `D`
      ( base
        `D`
      )
    )
    ( quality
      `la`
      ( solfege
        `la`
      )
    )
  )
  ( chord
    `Bbso`
    ( note
      `Bb`
      ( base
        `B`
      )
      ( accidental
        `b`
      )
    )
    ( quality
      `so`
      ( solfege
        `so`
      )
    )
  )
  ( chord
    `Absol`
    ( note
      `Ab`
      ( base
        `A`
      )
      ( accidental
        `b`
      )
    )
    ( quality
      `sol`
      ( solfege
        `sol`
      )
    )
  )
  ( chord
    `G#la`
    ( note
      `G#`
      ( base
        `G`
      )
      ( accidental
        `#`
      )
    )
    ( quality
      `la`
      ( solfege
        `la`
      )
    )
  )
  ( chord
    `A#ti`
    ( note
      `A#`
      ( base
        `A`
      )
      ( accidental
        `#`
      )
    )
    ( quality
      `ti`
      ( solfege
        `ti`
      )
    )
  )
  ( chord
    `Fmi`
    ( note
      `F`
      ( base
        `F`
      )
    )
    ( quality
      `mi`
      ( solfege
        `mi`
      )
    )
  )
)
